internal/webhook/webhook_utils_test.go

```diff
@@ -0,0 +1,460 @@
+	}{
+		{
+			name: "Has annotiation, Pod Affinity and Pod AntiAffinity",
+			pod: corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{
+						"leaderworkerset.sigs.k8s.io/exclusive-topology": "topologyKey",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Affinity: &corev1.Affinity{
+						PodAffinity: &corev1.PodAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{{TopologyKey: "topologyKey"}},
+						},
+						PodAntiAffinity: &corev1.PodAntiAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{{TopologyKey: "topologyKey"}},
+						},
+					},
+				},
+			},
+			expectedAppliedExclusivePlacement: true,
+		},
+		{
+			name: "Has annotiation, Pod Affinity, doesn't have Pod AntiAffinity",
+			pod: corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{
+						"leaderworkerset.sigs.k8s.io/exclusive-topology": "topologyKey",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Affinity: &corev1.Affinity{
+						PodAffinity: &corev1.PodAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{{TopologyKey: "topologyKey"}},
+						},
+					},
+				},
+			},
+			expectedAppliedExclusivePlacement: false,
+		},
+		{
+			name: "Has annotiation, Pod AntiAffinity, doesn't have Pod Affinity",
+			pod: corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{
+						"leaderworkerset.sigs.k8s.io/exclusive-topology": "topologyKey",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Affinity: &corev1.Affinity{
+						PodAntiAffinity: &corev1.PodAntiAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{{TopologyKey: "topologyKey"}},
+						},
+					},
+				},
+			},
+			expectedAppliedExclusivePlacement: false,
+		},
+		{
+			name: "Has annotiation, Pod Affinity and Pod AntiAffinity, Topology Key doesn't match",
+			pod: corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{
+						"leaderworkerset.sigs.k8s.io/exclusive-topology": "topologyKey",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Affinity: &corev1.Affinity{
+						PodAffinity: &corev1.PodAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{{TopologyKey: "topologyKey1"}},
+						},
+						PodAntiAffinity: &corev1.PodAntiAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{{TopologyKey: "topologyKey"}},
+						},
+					},
+				},
+			},
+			expectedAppliedExclusivePlacement: false,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			appliedExclusivePlacement := exclusiveAffinityApplied(tc.pod)
+			if appliedExclusivePlacement != tc.expectedAppliedExclusivePlacement {
+				t.Errorf("Expected value %t, got %t", tc.expectedAppliedExclusivePlacement, appliedExclusivePlacement)
+			}
+		})
+	}
+}
+
+func MakeLeaderPodSpec() corev1.PodSpec {
```

Comment:
can we add this and the functions below to the wrappers.go file and apply the wrapper pattern please?

Reply:
The testutils package imports the webhook package, so if I import testutils it creates an import cycle. That is why I needed to copy the functions into this file